### PR TITLE
Minimal fix for operation rename issue for HLC -> Modular client 

### PR DESCRIPTION
**Pull request by wanlwanl**

### CODE REVIEW

File: tools/js-sdk-release-tools/src/changelog/changelogGenerator.ts [hunk 1]
```diff
@@ -159,21 +159,36 @@ export class Changelog {
     }
 }
 
+// todo: special rules for HLC convert to Modular, will use a more generic method to replace
+function getRenamedOperationFromToMap(from: TSExportedMetaData, to: TSExportedMetaData):  {[id:string]: string} {
```
Comment (MaryGao) [inline tools/js-sdk-release-tools/src/changelog/changelogGenerator.ts:163]: Could you have a try with pr: https://github.com/Azure/azure-sdk-for-js/pull/29770?

what is the current changelog?

File: tools/js-sdk-release-tools/test/changelogGenerator.test.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { expect, test } from 'vitest'
+import { extractExportAndGenerateChangelog } from '../src/changelog/extractMetaData'
+import path from 'path';
+
+test('HLC -> Modular: Rename', async () => {
```
Comment (MaryGao) [inline tools/js-sdk-release-tools/test/changelogGenerator.test.ts:5]: Good to see we have testing here.

File: tools/js-sdk-release-tools/test/changelogGenerator.test.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  const oldViewPath = path.join(__dirname, "testCases/operationGroups.old.api.md");
+  const newViewPath = path.join(__dirname, "testCases/operationGroups.new.api.md");
+  const changelog = await extractExportAndGenerateChangelog(oldViewPath, newViewPath);
+  
+  expect(changelog.addedOperationGroup.length).toBe(0);
+  expect(changelog.removedOperationGroup.length).toBe(0);
+  expect(changelog.interfaceParamDelete.length).toBe(1);
+  expect(changelog.interfaceParamDelete[0]).toBe('Interface DataProductsCatalogsOperations no longer has parameter listByResourceXXXGroup');
```
Comment (MaryGao) [inline tools/js-sdk-release-tools/test/changelogGenerator.test.ts:13]: i notice the code has the change for operation, do you know what is the difference for operation and parameter here?

Comment (wanlwanl) [reply tools/js-sdk-release-tools/test/changelogGenerator.test.ts:13]: I'm not quite sure what you mean here. Could you provide more details? I didn't change the operations actually.

Comment (MaryGao) [reply tools/js-sdk-release-tools/test/changelogGenerator.test.ts:13]: Here is the comparation for operation: https://github.com/Azure/azure-sdk-tools/blob/7f1d4024c6de375b098f8074da055f0d3380555b/tools/js-sdk-release-tools/src/changelog/changelogGenerator.ts#L1040.

And in your change we have parameter change for listByResourceXXXGroup which is actually an operation.

File: tools/js-sdk-release-tools/src/changelog/changelogGenerator.ts (hunk 1 above)
Comment (MaryGao) [inline tools/js-sdk-release-tools/src/changelog/changelogGenerator.ts:163]: why we need `to` parameter here?

File: tools/js-sdk-release-tools/package.json [hunk 4]
```diff
@@ -7,7 +7,8 @@
     "start": "node dist/changelogToolCli.js",
     "debug": "node --inspect-brk dist/changelogToolCli.js",
     "build": "rimraf dist && tsc -p .",
-    "prepack": "npm run build"
+    "prepack": "npm run build",
```
Comment (MaryGao) [inline tools/js-sdk-release-tools/package.json:10]: bump a version?

File: tools/js-sdk-release-tools/test/changelogGenerator.test.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  expect(changelog.interfaceParamAddRequired.length).toBe(1);
+  expect(changelog.interfaceParamAddRequired[0]).toBe('Interface DataProductsCatalogsOperations has a new required parameter listByResourceGroup');
```
Comment (MaryGao) [inline tools/js-sdk-release-tools/test/changelogGenerator.test.ts:15]: Should be `Added operation DataProductsCatalogsOperations.listByResourceGroup`?
https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/monitor/arm-monitor/CHANGELOG.md#L617
image

File: tools/js-sdk-release-tools/test/changelogGenerator.test.ts (hunk 3 above)
Comment (wanlwanl) [reply tools/js-sdk-release-tools/test/changelogGenerator.test.ts]: updated

File: tools/js-sdk-release-tools/src/changelog/changelogGenerator.ts (hunk 1 above)
Comment (kazrael2119) [reply tools/js-sdk-release-tools/src/changelog/changelogGenerator.ts]: image
have code compile error in automaticGenerateChangeLogAndBumpVersion.ts with the latest commit

**Comment by wanlwanl** [comment]
@qiaozha @MaryGao check it out: https://github.com/Azure/azure-sdk-tools/actions/runs/9562789336/job/26371377608?pr=8436
From now on, changelog will be output in github check, so for review, @kazrael2119  no need to run test manually in the future

File: tools/js-sdk-release-tools/src/test/testCases/ci/arm-networkanalytics/recordings/node/networkanalytics_test/recording_dataproducts_delete_test.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+  "Entries": [],
```
Comment (MaryGao) [inline tools/js-sdk-release-tools/src/test/testCases/ci/arm-networkanalytics/recordings/node/networkanalytics_test/recording_dataproducts_delete_test.json:2]: no need for testing recording file

File: tools/js-sdk-release-tools/src/test/testCases/ci/arm-networkanalytics/.eslintrc.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
+{
+  "plugins": ["@azure/azure-sdk"],
+  "extends": ["plugin:@azure/azure-sdk/azure-sdk-base"],
```
Comment (MaryGao) [inline tools/js-sdk-release-tools/src/test/testCases/ci/arm-networkanalytics/.eslintrc.json:3]: could we remove useless files?

File: tools/js-sdk-release-tools/src/test/changelogGenerator.test.ts [hunk 8]
```diff
@@ -0,0 +1,94 @@
+import { expect, test } from "vitest";
+import { extractExportAndGenerateChangelog } from "../changelog/extractMetaData";
+import path from "path";
+import { SDKType } from "../common/types";
+
```
Comment (MaryGao) [inline tools/js-sdk-release-tools/src/test/changelogGenerator.test.ts:5]: Could you guide me on the case for networkanalytics?

Comment (MaryGao) [inline tools/js-sdk-release-tools/src/test/changelogGenerator.test.ts:5]: Could you guide me on the case for networkanalytics?[](url)

File: tools/js-sdk-release-tools/src/changelog/changelogGenerator.ts [hunk 9]
```diff
@@ -60,7 +62,7 @@ export class Changelog {
             this.typeAliasParamChangeRequired.length > 0 ||
             this.removedEnum.length > 0 ||
             this.removedEnumValue.length > 0;
-            this.removedFunction.length > 0;
+        this.removedFunction.length > 0;
```
Comment (MaryGao) [inline tools/js-sdk-release-tools/src/changelog/changelogGenerator.ts:65]: nit: format?

**Comment by raych1** [comment]
@wanlwanl , pls follow these steps to proceed the PR merge.
- resolve the comments 
- ask the code owner to approve
- approve the last stage of `azure-sdk-tools - sync - eng-workflows` pipeline
- once all the generated PR are merged, you can merge your PR

@MaryGao , the sync workflow was kicked off because this PR include the change on `.github` folder. It looks good to me as long as all the CI passes.

**Comment by weshaggard** [comment]
@wanlwanl @MaryGao can you please give me more info on this change? We cannot just create github actions in this repo as it has impacts on every other repo as we sync them to all repos.

**Comment by weshaggard** [comment]
Please remove the changes to the .github\workflows directory. I've reverted the PR https://github.com/Azure/azure-sdk/pull/7701 which should have never been merged and please do not merge any of the other sync PRs. Please close those once you undo the changes to the .github\workflows directory.

File: .github/workflows/js-sdk-release-tool-pull-request-check.yml [hunk 10]
```diff
@@ -0,0 +1,41 @@
+name: Javascript SDK Release Tool Pull Request Check
```
Comment (weshaggard) [inline .github/workflows/js-sdk-release-tool-pull-request-check.yml:1]: The github action needs to be reverted and we should continue to use the Devops pipeline ci.yml

**Review by weshaggard — CHANGES_REQUESTED** [review]
Please undo the workflow change and continue to use the devops ci.yml file.

File: .github/workflows/js-sdk-release-tool-pull-request-check.yml (hunk 10 above)
Comment (weshaggard) [reply .github/workflows/js-sdk-release-tool-pull-request-check.yml:1]: https://github.com/Azure/azure-sdk-tools/blob/main/tools/js-sdk-release-tools/ci.yml

Comment (wanlwanl) [reply .github/workflows/js-sdk-release-tool-pull-request-check.yml]: Thanks the guidance, removed

**Comment by wanlwanl** [comment]
@MaryGao  updated

**Review by MaryGao — APPROVED** [review]
LGTM

**Pull request merged**